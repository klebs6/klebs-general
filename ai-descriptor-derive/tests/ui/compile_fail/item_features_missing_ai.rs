#![allow(unused_imports)]
use ai_descriptor_derive::*;
use named_item::{ItemFeature, ItemWithFeatures};
use std::borrow::Cow;

#[derive(ItemWithFeatures)]
struct Potion {
    effect: PotionEffect,
}

struct PotionEffect;

impl ItemFeature for PotionEffect {
    fn text(&self) -> Cow<'_, str> {
        Cow::Borrowed("Gives you super strength.")
    }
}

fn main() {}

