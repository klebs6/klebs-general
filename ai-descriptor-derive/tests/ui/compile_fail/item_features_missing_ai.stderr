error: Structs deriving ItemWithFeatures must have an #[ai("...")] attribute for the header
 --> tests/ui/compile_fail/item_features_missing_ai.rs:7:8
  |
7 | struct Potion {
  |        ^^^^^^
